grails-app/controllers/org/pih/warehouse/requisition/RequisitionTemplateController.groovy

```diff
@@ -86,6 +86,79 @@ class RequisitionTemplateController {
         }
     }
 
+    def exportPage = {
+        render(view:"exportPage")
+    }
+
+    def exportAsCsv = {
+        def requisitionItems = []
+        def origins = []
+        def destinations = []
+
+        params.origin.each { origin ->
+            origins << Location.get(origin)
+        }
+        params.destination.each { destination ->
+            destinations << Location.get(destination)
+        }
+
+        def requisitions = requisitionService.getRequisitionTemplates(origins, destinations)
+        requisitionService.getRequisitionTemplatesItems(requisitions).groupBy { it.product }.each { product, value ->
+            requisitionItems << [
+                    product: product,
+                    productCode: product.productCode ?: "",
+                    productName: product.name ?: "",
+                    category: product?.category?.name ?: "",
+                    pricePerUnit: product?.pricePerUnit ?: "",
+            ]
+        }
+
+        def sw = new StringWriter()
+
+        try {
+            if (requisitionItems) {
+                sw.append("Item code").append(",")
+                sw.append("Item description").append(",")
+                sw.append("Category").append(",")
+
+               requisitions.each { requisition ->
+                    String stocklistName = StringEscapeUtils.escapeCsv(requisition?.name)
+                    sw.append(stocklistName).append(",")
+                }
+
+                sw.append("Unit cost").append(",")
+                sw.append("Total value")
+
+                sw.append("\n")
+                requisitionItems.each { requisitionItem ->
+                    def totalCost = 0
+                    requisitions.each { requisition ->
+                        totalCost = requisitionItem.pricePerUnit ? totalCost + (requisition.getQuantityByProduct(requisitionItem.product) * requisitionItem.pricePerUnit) : ""
+                    }
+
+                    sw.append('"' + (requisitionItem.productCode).toString()?.replace('"','""') + '"').append(",")
```

Comment:
I mentioned in a previous PR that we should never ever do this even if what you think should be working doesn't work. Always use StringEscapeUtils to escape strings. The problem you're encountering here is probably because you're not using string writer properly. Using a plus operation within append leads to add behavior. Please also don't do that.

The following should work. 
```
sw.append(StringEscapeUtils.escapeCsv(requisitionItem?.productCode)).append(",")
```